daemon/graphdriver/lcow/lcow_svm.go

```diff
@@ -0,0 +1,394 @@
+// +build windows
+
+package lcow
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"io"
+
+	"github.com/Microsoft/hcsshim"
+	"github.com/jhowardmsft/opengcs/gogcs/client"
+	"github.com/sirupsen/logrus"
+)
+
+// Code for all the service VM management for the LCOW graphdriver
+
+var errVMisTerminating = errors.New("service VM is shutting down")
+var errVMUnknown = errors.New("service vm id is unknown")
+var errVMStillHasReference = errors.New("Attemping to delete a VM that is still being used")
+
+// serviceVMMap is the struct representing the id -> service VM mapping.
+type serviceVMMap struct {
+	sync.Mutex
+	svms map[string]*serviceVMWithRef
+}
+
+// serviceVMWithRef is our internal structure representing an item in our
+// map of service VMs we are maintaining.
+type serviceVMWithRef struct {
+	svm      *serviceVM // actual service vm object
+	refCount int        // refcount for VM
+}
+
+type serviceVM struct {
+	sync.Mutex                     // Serialises operations being performed in this service VM.
+	scratchAttached bool           // Has a scratch been attached?
+	config          *client.Config // Represents the service VM item.
+
+	// Indicate that the vm is started
+	startStatus chan interface{}
+	startError  error
+
+	// Indicates that the vm is stopped
+	stopStatus chan interface{}
+	stopError  error
+
+	// NOTE: It is OK to use a cache here because Windows does not support
+	// restoring containers when the daemon dies.
+	attachedVHDs map[string]int // Map ref counting all the VHDS we've mounted/unmounted.
+	unionMounts  map[string]int // Map ref counting all the union filesystems we mounted.
+}
+
+func newServiceVMMap() *serviceVMMap {
```

Comment:
Wondering if it's worth having this constructor rather than inlining in driver init?